adam/learner/attributes.py

```diff
@@ -98,7 +99,25 @@ def _candidate_templates(
                         )
                     )
 
-        return immutableset(ret)
+        return immutableset(
+            bound_surface_template
+            for bound_surface_template in ret
+            # For now, we require templates to account for the entire utterance.
+            # See https://github.com/isi-vista/adam/issues/789
+            # We strictly ignore english determiners here for determining
+            # the length needed of our surface template
+            if covers_entire_utterance(
+                bound_surface_template,
+                language_concept_alignment,
+                token_sequence_count=len(
+                    [
+                        token
+                        for token in language_concept_alignment.language.as_token_sequence()
+                        if token not in ["a", "the"]
```

Comment:
@Reviewer - Any ideas why the attribute learner needs this specific condition to be able to learn color attributes while the relationship and verb learner can handle the determiners?

Reply:
This was found as a result of the attribute SurfaceTemplate not appropriately adding the SLOT variables as having determiners.